infrastructure/db/handler.js

```diff
@@ -1,253 +1,208 @@
 * @return {Promise} promise, with json response
 */
-exports.getFolders = async(teammemeberId) => {
-
-    return new Promise((resolve) => {
-
-        models.folder.findAll({ where : { teammemeberId : teammemeberId} }).then(folders => {
-
-            let resp = { folders : new Array() };
-            for(let i = 0; i < folders.length;i++) {
-                resp.folders[i] = {
-                    id : folders[i].id,
-                    label : folders[i].label
-                }
-            }
-            resolve(resp);
-
-        });
-
-    });
-}
+const getFolders = async(teammemberId) => {
+    try {
+        return await model.folder.findAll({ where : { teammemberId : teammemberId} });
+    } catch(err) {
+        throw Error(err);
+    }
+};
 
 /**
 * Adds folder to folders table with corresponding given information
 * @param {json} request post body request
-* @return Nothing
+* @return Nothing.
 */
-exports.addFolder = async(request) => {
-    models.folder.create({ label : request.label, teammemberId : request.id});
+const addFolder = async(request) => {
+    try {
+        models.folder.create({ label : request.label, teammemberId : request.id});
+    } catch(err) {
+        throw Error(err);
+    }
 }
 
 /**
 * Finds all the boards of specific Team member
-* @param {int} teammemeberId id of the wanted teammember
-* @return {Promise} promise, with json response
+* @param {int} teammemberId id of the wanted teammember
+* @return {JSON} Json object with board detailss
 */
-exports.getBoards = async (teammemeberId) => {
+const getBoards = async(teammemberId) => {
+    try {
+        const teammember = await getTeamMember(teammemberId);
+        return await teammember.getBoards();
+    } catch(err) {
+        throw Error(err);
+    }
+};
 
-    return new Promise((resolve) =>  {
-        models.teammember.findByPk(teammemeberId).then(user => {
-        
-            user.getBoards().then(boards => {
-                let response = {
-                    boards : new Array()
-                }
-                for(let i = 0; i < boards.length;i++) {
-                    response.boards[i] = {
-                        id : boards[i].id,
-                        label : boards[i].label
-                    }
-                }
-
-                resolve(response);
-
-            });
-        });
-    });
-}
-
-exports.addBoard = async(request) => {
-
-    models.teammember.findByPk(request.id).then(user => {
-        models.board.create({ label : request.label, folderId: request.folder}).then(board => {
-            board.setTeammembers(user, {});
-        });
-    });
-    
+/**
+* Creates new Board
+* @param {JSON} board Details of the new Board
+* @return Nothing.
+*/
+const addBoard = async(board) => {
+    try {
+        const board = await models.board.create({ label : request.label, folderId: request.folder});
+        const member = await getTeamMember(board.teammemberId);
+        board.setTeammembers(member, {});
```

Comment:
because you are applying the `setTeamMembers` method on the `board` resource you need to apply `await` here also because you are talking to the DB and it's an async operation

Reply:
by the way, IMO `setTeamMembers` is more readable than `setTeammembers`